pytorch_lightning/strategies/launchers/multi_process.py

```diff
@@ -0,0 +1,112 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import subprocess
+import sys
+from subprocess import Popen
+from time import sleep
+from typing import Any, Callable, List, Optional
+
+import __main__
+import numpy as np
+
+from pytorch_lightning.plugins.environments.cluster_environment import ClusterEnvironment
+from pytorch_lightning.strategies.launchers.base import Launcher
+from pytorch_lightning.utilities import _HYDRA_AVAILABLE
+
+if _HYDRA_AVAILABLE:
+    from hydra.core.hydra_config import HydraConfig
+    from hydra.utils import get_original_cwd, to_absolute_path
+
+
+class MultiProcessLauncher(Launcher):
```

Comment:
The name is too close to `multiprocessing.Process`, but we actually use the subprocess module. I would suggest to choose a different name. 

- SubprocessLauncher
- SubprocessScriptLauncher (to emphasize this calls the script directly)

Reply:
I vote for the second one although it is quite long :D

Reply:
done!